firestore/src/main/local_cache_settings_main.h

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIREBASE_FIRESTORE_SRC_MAIN_LOCAL_CACHE_SETTINGS_MAIN_H_
+#define FIREBASE_FIRESTORE_SRC_MAIN_LOCAL_CACHE_SETTINGS_MAIN_H_
+
+#include <cstdint>
+#include <utility>
+
+#include "Firestore/core/src/api/settings.h"
+#include "firebase/firestore/local_cache_settings.h"
+
+namespace firebase {
+namespace firestore {
+
+class LocalCacheSettingsInternal {
+ public:
+  friend bool operator==(const LocalCacheSettingsInternal& lhs,
+                         const LocalCacheSettingsInternal& rhs) {
+    return &lhs == &rhs || lhs.core_settings() == rhs.core_settings();
```

Comment:
Is this checking for `&lhs == &rhs` strictly required? If no, I recommend removing it since it is a performance optimization that likely is not warranted. Applies here and elsewhere in this file.